src/cli_plugin/install/zip.js

```diff
@@ -47,31 +50,32 @@ export function analyzeArchive(archive) {
           return zipfile.readEntry();
         }
 
-        zipfile.openReadStream(entry, function (err, readable) {
+        zipfile.openReadStream(entry, function (error, readable) {
           const chunks = [];
 
-          if (err) {
-            return reject(err);
+          if (error) {
+            return reject(error);
           }
 
           readable.on('data', (chunk) => chunks.push(chunk));
 
           readable.on('end', function () {
-            const contents = Buffer.concat(chunks).toString();
-            const pkg = JSON.parse(contents);
-
-            plugins.push(
-              Object.assign(pkg, {
-                archivePath: match[1],
-                archive: archive,
-
-                // Plugins must specify their version, and by default that version should match
-                // the version of kibana down to the patch level. If these two versions need
-                // to diverge, they can specify a kibana.version to indicate the version of
-                // kibana the plugin is intended to work with.
-                kibanaVersion: get(pkg, 'kibana.version', pkg.version),
-              })
-            );
+            const manifestJson = Buffer.concat(chunks).toString();
+            const manifest = JSON.parse(manifestJson);
+
+            plugins.push({
+              id: manifest.id,
+              stripPrefix: match[1],
+
+              // Plugins must specify their version, and by default that version in the plugin
+              // manifest should match the version of kibana down to the patch level. If these
+              // two versions need plugins can specify a kibanaVersion to indicate the version
+              // of kibana the plugin is intended to work with.
+              kibanaVersion:
+                typeof manifest.kibanaVersion === 'string' && manifest.kibanaVersion
+                  ? manifest.kibanaVersion
+                  : manifest.version,
```

Comment:
I think we should only use the `kibanaVersion` key and ignore the plugin version. I'd prefer we err on the side of explicitness here and force plugins to declare their intention. If kibanaVersion is not specified, we should throw an error.

We should also adjust the logic in [here in Core](https://github.com/spalger/kibana/blob/a96979ff475e4c62dc39c80f27a2575d34e6aac2/src/core/server/plugins/discovery/plugin_manifest_parser.ts#L152) to match. This could be done as a follow up.

Reply:
Yeah, I based this logic off of that logic but I welcome that change as a follow up.